transport/tchannel/handler.go

```diff
@@ -243,7 +243,7 @@ func (rw *responseWriter) Close() error {
 			retErr = appendError(retErr, fmt.Errorf("SetApplicationError() failed: %v", err))
 		}
 	}
-	retErr = appendError(retErr, writeHeaders(rw.format, rw.headers, rw.response.Arg2Writer))
+	retErr = appendError(retErr, writeHeaders(rw.format, rw.headers.Items(), rw.response.Arg2Writer))
```

Comment:
Shouldn't the exact case option matter here?

Reply:
no this is the responseWriter. The change here only cares about outbound requests.

Reply:
I think we have the exact same issue with response headers here though.. It seems odd that only request headers are kept in the exact case, but response headers are always canonicalized?